### PR TITLE
Wiki integration

### PR DESCRIPTION
**Pull request by inciotiq**

### CODE REVIEW

File: src/main/java/com/iotiq/application/wiki/WikiAuthServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        }
+    }
+
+    @Override
+    public String getAccessToken() {
+        return "Bearer " + auth.getAuthToken();
+    }
+
+    private void authenticate() {
+        String username = "inci@iotiq.net";
```
Comment (oguzIotiq) [inline src/main/java/com/iotiq/application/wiki/WikiAuthServiceImpl.java:44]: move to application-dev.yml

File: src/main/java/com/iotiq/application/wiki/WikiClientImpl.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+@Service
+public class WikiClientImpl implements WikiClient {
+
+    private final WikiAuthService authService;
+    private final HttpGraphQlClient graphQlClient;
+
+    public WikiClientImpl(
+            WikiAuthService wikiAuthService,
+            @Value("http://localhost/graphql/") String url
```
Comment (oguzIotiq) [inline src/main/java/com/iotiq/application/wiki/WikiClientImpl.java:26]: to application-dev.yml

File: src/main/java/com/iotiq/application/wiki/domain/WikiAuth.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    public static WikiAuth from(WikiAuthResponse authResponse) {
+        String authToken = authResponse.getJwt();
+        Date accessExpirationTime = getExpirationTime(authToken);
+        return WikiAuth.builder()
+                .authToken(authToken)
+                .accessTokenValidUntil(accessExpirationTime)
+                .build();
+    }
+
+    private static Date getExpirationTime(String token) {
```
Comment (oguzIotiq) [inline src/main/java/com/iotiq/application/wiki/domain/WikiAuth.java:32]: can easily be a method of JWTUtils

File: src/main/java/com/iotiq/application/wiki/domain/WikiAuth.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        try {
+            JWT parse = JWTParser.parse(token);
+            return parse.getJWTClaimsSet().getExpirationTime();
+        } catch (ParseException e) {
+            logger.error("Could not parse expiration date from token");
+        }
+        return Date.from(Instant.now());
+    }
+
+    public boolean isAccessTokenExpired() {
```
Comment (oguzIotiq) [inline src/main/java/com/iotiq/application/wiki/domain/WikiAuth.java:42]: This is also.

File: src/main/java/com/iotiq/application/wiki/messages/WikiAuthResponse.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
+package com.iotiq.application.wiki.messages;
+
+import lombok.Data;
+
+@Data
```
Comment (oguzIotiq) [inline src/main/java/com/iotiq/application/wiki/messages/WikiAuthResponse.java:5]: Consider this to be a record

File: src/main/java/com/iotiq/application/wiki/messages/ResponseResult.java [hunk 6]
```diff
@@ -0,0 +1,11 @@
+package com.iotiq.application.wiki.messages;
+
+import lombok.Data;
+
+@Data
```
Comment (oguzIotiq) [inline src/main/java/com/iotiq/application/wiki/messages/ResponseResult.java:5]: Consider this to be a record

File: src/main/java/com/iotiq/application/wiki/domain/PageDto.java [hunk 7]
```diff
@@ -0,0 +1,10 @@
+package com.iotiq.application.wiki.domain;
+
+import lombok.Data;
+
+@Data
+public class PageDto {
```
Comment (oguzIotiq) [inline src/main/java/com/iotiq/application/wiki/domain/PageDto.java:6]: Consider this to be a record

File: src/main/java/com/iotiq/application/wiki/WikiAuthServiceImpl.java (hunk 1 above)
Comment (inciotiq) [reply src/main/java/com/iotiq/application/wiki/WikiAuthServiceImpl.java]: moved

File: src/main/java/com/iotiq/application/wiki/WikiClientImpl.java (hunk 2 above)
Comment (inciotiq) [reply src/main/java/com/iotiq/application/wiki/WikiClientImpl.java]: moved

File: src/main/java/com/iotiq/application/wiki/domain/WikiAuth.java (hunk 3 above)
Comment (inciotiq) [reply src/main/java/com/iotiq/application/wiki/domain/WikiAuth.java:32]: I think it makes sense here

File: src/main/java/com/iotiq/application/wiki/messages/WikiAuthResponse.java (hunk 5 above)
Comment (inciotiq) [reply src/main/java/com/iotiq/application/wiki/messages/WikiAuthResponse.java]: changed to record

File: src/main/java/com/iotiq/application/wiki/messages/ResponseResult.java (hunk 6 above)
Comment (inciotiq) [reply src/main/java/com/iotiq/application/wiki/messages/ResponseResult.java]: changed to record

File: src/main/java/com/iotiq/application/wiki/domain/PageDto.java (hunk 7 above)
Comment (inciotiq) [reply src/main/java/com/iotiq/application/wiki/domain/PageDto.java]: changed to record

**Pull request merged**